hashing/te-tag-query-python/TE.py

```diff
@@ -293,6 +293,79 @@ def getInfoForIDs(self, ids, **kwargs):
 
         return descriptors
 
+    # ----------------------------------------------------------------
+    # See also https://developers.facebook.com/docs/threat-exchange/reference/apis/threat-descriptors
+    #
+    # NOTE: THIS IS KNOWN TO NOT CORRECTLY HANDLE PAGINATION.  Please use the
+    # tagged_objects endpoint as detailed above.
+
+    @classmethod
+    def doPowerSearch(self, descriptorBatchProcessorCallback, urlParams, options):
+        verbose = options.get("verbose", False)
+        showURLs = options.get("showURLs", False)
+        includeIndicatorInOutput = options.get("includeIndicatorInOutput", True)
+
+        startURL = (
+            self.TE_BASE_URL
+            + "/threat_descriptors"
+            + "/?access_token="
+            + self.APP_TOKEN
+            + "&fields=raw_indicator,type,added_on,last_updated,confidence,owner,privacy_type,review_status,status,severity,share_level,tags,description,reactions,my_reactions"
+        )
```

Comment:
(ignorable, alt): "/".join() might be as an alternative
(ignorable, alt): check out urlencode (https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) for the query as a possibility, which can handle formatting and escaping